### DISCUSSION TITLE
Internal VM storage usage

### DISCUSSION THREAD

**Discussion by aDingil** [kube-hetzner/terraform-hcloud-kube-hetzner#787] [Q&A] [answered]
Is it possible to use the internal storage space, instead of volumes?

**Comment by mysticaltech** [comment, marked as answer]
@aDingil Most definitely, just set the volume size to 0, see kube.tf agent nodepool section for more details.

**Reply by aDingil** [reply]
So internal volume usage only applies to longhorn?
On the other hand using internal volumes for nodes doesn´t make that much sense.

**Reply by mysticaltech** [reply]
To clarify, with Longhorn, each node can be attached to one volume, OR use its on storage. I'm also seen users use both (search this project for `Longhorn`, especially look for the best practice guide in the results).

When using Hetzner CSI, the only option you have is to use volumes provided by it.

**Comment by aDingil** [comment]
Thx
